### DISCUSSION TITLE
Necessary to run `yarn` for local development to install / set up every package?

### DISCUSSION THREAD

**Discussion by john-b-yang** [cypress-io/cypress#29976] [Questions and Help]
Hello! I am a new here and interested in contributing, so apologies in advance if I overlooked some documentation.

I am primarily interested in contributing to `packages/driver`. However, from the contributing guidelines, I noticed that the command to install every package upon cloning locally is to run `yarn`.

I was wondering, is there an equivalent to this, but just for setting up a single package? My laptop is a bit slow, and running `yarn` to install everything sometimes takes a lot of time.

In my case, I'm interested in just working on `packages/driver`. Is there a command for just running installation for that package? Would the command, if it exists, look similar for other packages such as `server` or `app`?

**Comment by portyu9** [comment]
Hi @john-b-yang 

Yes — for the Cypress repository, the supported development setup is to run `yarn` from the **repository root**.

The repo is a Yarn-workspaces monorepo and the contribution guide explicitly says:

- install dependencies from the root;
- do not install dependencies or run scripts from a package subdirectory;
- use scoped tasks when you only want to work on one or a few packages.

So the initial setup is:

```sh
yarn
```

There is not a supported equivalent that installs only `packages/driver` and skips the rest of the workspace setup.

After that, you **can** greatly reduce the work performed by scoping commands to the package you are changing. For example:

```sh
yarn test-unit --scope @packages/driver
```

or multiple related packages:

```sh
yarn test-unit \
  --scope @packages/driver \
  --scope @packages/server
```

The same `--scope` mechanism is available for many of the top-level tasks such as `build`, `watch`, and the various test tasks.

That is the intended compromise: perform the monorepo dependency setup once, then scope development/build/test work to the packages relevant to your change.

Also check the README for the individual package you are modifying because package-specific development/test instructions can differ.

No screenshots or attachments were included in this discussion.

Please mark this answer as accepted if it helped, thank you.
